shared/lib_battery_powerflow.cpp

```diff
@@ -154,358 +154,401 @@ double BatteryPower::adjustForDCEfficiencies(double power) {
-
-	// Quantities are DC in KW unless otherwise specified
-	double P_pv_to_batt_dc, P_grid_to_batt_dc, P_pv_to_inverter_dc;
-	P_pv_to_batt_dc = P_grid_to_batt_dc = P_pv_to_inverter_dc = 0;
-
-	// The battery power and PV power are initially DC, which must be converted to AC for powerflow
-	double P_battery_dc_pre_bms = m_BatteryPower->powerBatteryDC;
-	double P_battery_dc = m_BatteryPower->powerBatteryDC;
-	double P_pv_dc = m_BatteryPower->powerPV;
-
-	// convert the calculated DC power to DC at the PV system voltage
-	if (P_battery_dc_pre_bms < 0)
-		P_battery_dc = P_battery_dc_pre_bms / m_BatteryPower->singlePointEfficiencyDCToDC;
-	else if (P_battery_dc > 0)
-		P_battery_dc = P_battery_dc_pre_bms * m_BatteryPower->singlePointEfficiencyDCToDC;
-
-	double P_gen_dc = P_pv_dc + P_battery_dc;
-
-	// in the event that PV system isn't operating, assume battery BMS converts battery voltage to nominal inverter input at the weighted efficiency
-	double voltage = m_BatteryPower->voltageSystem;
-	double efficiencyDCAC = m_BatteryPower->sharedInverter->efficiencyAC * 0.01;
-	if (voltage <= 0) {
-		voltage = m_BatteryPower->sharedInverter->getInverterDCNominalVoltage();
-	}
-	if (std::isnan(efficiencyDCAC) || m_BatteryPower->sharedInverter->efficiencyAC <= 0) {
-		efficiencyDCAC = m_BatteryPower->sharedInverter->getMaxPowerEfficiency() * 0.01;
-	}
-
-
-	// charging
-	if (P_battery_dc < 0)
-	{
-		// First check whether battery charging came from PV.
-		// Assumes that if battery is charging and can charge from PV, that it will charge from PV before using the grid
-		if (m_BatteryPower->canPVCharge || m_BatteryPower->canClipCharge) {
-			P_pv_to_batt_dc = fabs(P_battery_dc);
-			if (P_pv_to_batt_dc > P_pv_dc) {
-				P_pv_to_batt_dc = P_pv_dc;
-			}
-		}
-		P_pv_to_inverter_dc = P_pv_dc - P_pv_to_batt_dc;
-
-		// Any remaining charge comes from grid if allowed
-		P_grid_to_batt_dc = fabs(P_battery_dc) - P_pv_to_batt_dc;
-        if (!m_BatteryPower->canGridCharge && P_grid_to_batt_dc > tolerance){
+    // Quantities are AC in KW unless otherwise specified
+    double P_load_ac = m_BatteryPower->powerLoad;
+    double P_battery_ac, P_pv_ac, P_gen_ac, P_pv_to_batt_ac, P_grid_to_batt_ac, P_batt_to_load_ac, P_grid_to_load_ac, P_pv_to_load_ac, P_pv_to_grid_ac, P_batt_to_grid_ac, P_grid_ac, P_conversion_loss_ac;
+    P_battery_ac = P_pv_ac = P_pv_to_batt_ac = P_grid_to_batt_ac = P_batt_to_load_ac = P_grid_to_load_ac = P_pv_to_load_ac = P_pv_to_grid_ac = P_batt_to_grid_ac = P_gen_ac = P_grid_ac = P_conversion_loss_ac = 0;
+
+    // Quantities are DC in KW unless otherwise specified
+    double P_pv_to_batt_dc, P_grid_to_batt_dc, P_pv_to_inverter_dc;
+    P_pv_to_batt_dc = P_grid_to_batt_dc = P_pv_to_inverter_dc = 0;
+    double P_system_loss_dc = m_BatteryPower->powerSystemLoss; // Units of power sytem loss match battery connection type, see help for details
```

Comment:
Okay, I see this now-- can you add this to the var_info tables for the compute modules?